modules/seqexec/server/src/test/scala/seqexec/server/SeqexecEngineSpec.scala

```diff
@@ -170,221 +175,269 @@ class SeqexecEngineSpec extends FlatSpec with Matchers {
   private val seqId2 = "GS-2018B-Q-0-2"
   private val seqObsId2 = Observation.Id.unsafeFromString(seqId2)
   private val seqId3 = "GS-2018B-Q-0-3"
   private val seqObsId3 = Observation.Id.unsafeFromString(seqId3)
-  private def sequence(id: Observation.Id): Sequence.State = Sequence.State.init(
-    Sequence(
-      id,
-      SequenceMetadata(Instrument.F2, None, ""),
-      List(Step.init(1, None, Map(), Set(), List(List(pendingAction(Instrument.F2)))))
-    )
-  )
+  private def sequence(id: Observation.Id): SequenceGen = SequenceGen(
+    id,
+    "",
+    Instrument.F2,
+    List(SequenceGen.Step(1, Map(), Set.empty, _ => Step.init(1, None, List(List(pendingAction(Instrument.F2)))))))
+
   "SeqexecEngine addSequenceToQueue" should
     "add sequence id to queue" in {
-    val s0 = Engine.State.empty(EngineMetadata.default)
-      .copy(sequences = Map(seqObsId1 -> sequence(seqObsId1)))
+    val s0 = SeqexecEngine.loadSequenceEndo(seqObsId1, sequence(seqObsId1))(EngineState.default)
 
     (for {
-      q <- Stream.eval(async.boundedQueue[IO, executeEngine.EventType](10))
+      q <- async.boundedQueue[IO, executeEngine.EventType](10)
       sf <- advanceOne(q, s0, seqexecEngine.addSequenceToQueue(q, CalibrationQueueName, seqObsId1))
     } yield {
-      inside(sf.flatMap(x => EngineMetadata.queues.get(x.userData).get(CalibrationQueueName))) {
+      inside(sf.flatMap(x => EngineState.queues.get(x).get(CalibrationQueueName))) {
         case Some(exq) => exq shouldBe List(seqObsId1)
       }
-    }).compile.last.unsafeRunSync
+    }).unsafeRunSync
   }
   it should "not add sequence id if sequence does not exists" in {
     val badObsId = Observation.Id.unsafeFromString("NonExistent-1")
-    val s0 = Engine.State.empty(EngineMetadata.default)
-      .copy(sequences = Map(seqObsId1 -> sequence(seqObsId1)))
+    val s0 = SeqexecEngine.loadSequenceEndo(seqObsId1, sequence(seqObsId1))(EngineState.default)
 
     (for {
-      q <- Stream.eval(async.boundedQueue[IO, executeEngine.EventType](10))
+      q <- async.boundedQueue[IO, executeEngine.EventType](10)
       sf <- advanceOne(q, s0, seqexecEngine.addSequenceToQueue(q, CalibrationQueueName, badObsId))
     } yield {
-      inside(sf.flatMap(x => EngineMetadata.queues.get(x.userData).get(CalibrationQueueName))) {
+      inside(sf.flatMap(x => EngineState.queues.get(x).get(CalibrationQueueName))) {
         case Some(exq) => assert(exq.isEmpty)
       }
-    }).compile.last.unsafeRunSync
+    }).unsafeRunSync
   }
+
   it should "not add sequence id if sequence is running or completed" in {
-    val s0 = Engine.State.empty(EngineMetadata.default)
-      .copy(sequences = Map(seqObsId1 -> Sequence.State.status.set(SequenceState.Running.init)(sequence(seqObsId1)),
-        seqObsId2 -> Sequence.State.status.set(SequenceState.Completed)(sequence(seqObsId2))
-      ) )
+    val s0 = (SeqexecEngine.loadSequenceEndo(seqObsId1, sequence(seqObsId1)) >>>
```

Comment:
is `>>>` the same as `andThen` right? is that on the core lib or from cats?

Reply:
Yes, it is the same, and it is from cats.